backends/dpdk/dpdkAsmOpt.h

```diff
@@ -159,6 +159,121 @@ class RemoveUnusedMetadataFields : public Transform {
     const IR::Node* preorder(IR::DpdkAsmProgram *p) override;
 };
 
+// This pass shorten the Identifier length
+class ShortenTokenLength : public Transform {
+    ordered_map<cstring, cstring> newNameMap;
+    ordered_map<cstring, cstring> &origNameMap;
+    static size_t count;
+    // Currently Dpdk allows Identifier of 63 char long or less
+    // including dots(.) for member exp.
+    // worst case member expression will look like below(for headers)
+    // 1.30.30 => 63(including dot(.))
+    cstring shortenString(cstring str, size_t allowedLength = 60) {
+        // if id name less than allowedLength keep it same
+        if (str.size() <= allowedLength)
+            return str;
+        auto itr = newNameMap.find(str);
+        if (itr != newNameMap.end())
+            return itr->second;
+        // make sure new string length less or equal allowedLength
+        cstring newStr = str.substr(0, allowedLength - std::to_string(count).size());
+        newStr += std::to_string(count);
+        count++;
+        newNameMap.insert(std::pair<cstring, cstring>(str, newStr));
+        origNameMap.insert(std::pair<cstring, cstring>(newStr, str));
+        return newStr;
+    }
+
+ public:
+    explicit ShortenTokenLength(ordered_map<cstring, cstring> &origNameMap)
+        : origNameMap{origNameMap} {}
+    const IR::Node* preorder(IR::Member *m) override {
+        if (m->toString().startsWith("h."))
+            m->member = shortenString(m->member, 30);
+        else
+            m->member = shortenString(m->member);
+        return m;
+    }
+
+    const IR::Node* preorder(IR::DpdkStructType *s) override {
+        for (auto annot : s->annotations->annotations)
+            if (annot->name == "__packet_data__") {
+                s->name = shortenString(s->name);
+                for (auto field : s->fields) {
+                    IR::StructField *f = const_cast<IR::StructField *>(field);
+                    f->name = shortenString(f->name, 30);
+                    if (auto st = f->to<IR::DpdkStructType>()) {
+                        for (auto field1 : st->fields) {
+                             IR::StructField *f1 = const_cast<IR::StructField *>(field1);
```

Comment:
const_cast is discouraged 
https://github.com/p4lang/p4c/blob/main/docs/README.md#coding-conventions

You can try creating a copy and updating it.